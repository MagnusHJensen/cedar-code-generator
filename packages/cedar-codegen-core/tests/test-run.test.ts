import { main } from '@cedar-codegen/core';
import { readFileSync } from 'node:fs';
import { join } from 'node:path';

test('matches snapshot', () => {
  main({
    output: join(__dirname, '/test-files/cedar-types.ts'),
    schema: join(__dirname, '/test-files/test.cedarschema.json')
  });

  expect(
    readFileSync(join(__dirname, '/test-files/cedar-types.ts')).toString()
  ).toMatchSnapshot();
});
