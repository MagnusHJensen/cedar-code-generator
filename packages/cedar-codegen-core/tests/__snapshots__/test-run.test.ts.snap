// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
"export const schemaNamespace: string = "MyNamespace";

export type PhotoActions = "viewPhoto";

export type CedarActions = "noResourceAction" | PhotoActions;

export enum CedarPrincipal {
    USER = "User"
}

export enum CedarResource {
    PHOTO = "Photo"
}

interface UserNestedData {
    nestedName: string;
}

interface UserData {
    principalType: CedarPrincipal.USER;
    name: string;
    isActive: boolean;
    age?: number;
    nested: UserNestedData;
}

interface PhotoData {
    principalType: CedarResource.PHOTO;
    url?: string;
    createdAt: string;
}

type CedarPrincipalDataTypes = UserData;

type CedarResourceDataTypes = PhotoData;

/**
 * Helper methods to prefix actions and enums with namespace */

export function prefixEnumWithNamespace(namespace: CedarResource | CedarPrincipal): string { return schemaNamespace + "::" + namespace; }

export function prefixActionWithNamespace(action: CedarActions): string { return schemaNamespace + "::Action::" + action; }

"
`;
